pallets/exchange/benchmarking/src/lib.rs

```diff
@@ -144,33 +160,55 @@ benchmarks! {
 		let amount : Balance = DOLLARS;
 		let limit : Balance = DOLLARS;
 
+		let nbr_intentions_appended: u32  = MAX_INTENTIONS_IN_BLOCK;
+
 		initialize_pool::<T>(caller.clone(), asset_a, asset_b, amount, Price::from(1))?;
 
-		assert_eq!(pallet_exchange::Module::<T>::get_intentions_count((asset_a, asset_b)), 0);
+		let random_seed = BlakeTwo256::hash(b"Sixty-nine");
+		let mut rng = <RandomNumberGenerator<BlakeTwo256>>::new(random_seed);
+		let mut amounts: Vec<u32> = Vec::with_capacity(nbr_intentions_appended as usize);
+		for idx in 0 .. nbr_intentions_appended {
+			amounts.push(rng.pick_u32(100_000) + 1000);
+		}
+
+		feed_intentions::<T>(asset_a, asset_b, nbr_intentions_appended, &amounts)?;
+
+		assert_eq!(pallet_exchange::Module::<T>::get_intentions_count((asset_a, asset_b)), nbr_intentions_appended);
 
 	}: {  Exchange::<T>::buy(RawOrigin::Signed(caller.clone()).into(), asset_a, asset_b, amount / 10 ,limit, false)? }
 	verify{
-		assert_eq!(pallet_exchange::Module::<T>::get_intentions_count((asset_a, asset_b)), 1);
+		assert_eq!(pallet_exchange::Module::<T>::get_intentions_count((asset_a, asset_b)), nbr_intentions_appended + 1);
 	}
 
 	on_finalize {
-		let t in 0 .. 100; // Intention component
+		let t in 0 .. MAX_INTENTIONS_IN_BLOCK; // Intention component
 		let caller = funded_account::<T>("caller", 1);
 
 		let asset_a: AssetId = 1;
 		let asset_b: AssetId = 2;
 		let amount : Balance = 100_000_000_000_000;
 
+		// First generate random amounts
+		// This is basically used to generate intentions with different amounts
+		// it is because algorithm does sort the intention by amount, so we need something not sorted./
```

Comment:
Is it possible to assume worst case scenario here better than randomness?

Reply:
Worst case would be here the "unsorted list of amounts". That's what the test case does. It is just creates list of random amounts, this would be presumably unsorted. 
CAn you of other way ?

Reply:
I feel like random is not the best way to benchmark as it can create different results with different levels of unsortedness. We should generate an arbitrary unsorted sequence that we will input every time? Maybe to external file if big.

Reply:
Ok. Make sense. It should be ok - the size is same as max intention.